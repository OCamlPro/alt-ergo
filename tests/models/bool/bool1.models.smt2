(set-logic QF_UF)
(set-option :produce-models true) ; enable model generation
(set-option :produce-assignments true) ; enable get assignment
(declare-const p Bool)
(declare-const q Bool)
; (declare-const t Int)
(define-fun nq () Bool q)
(assert (=> (not p) (not nq)))
(check-sat)
(get-model)
(get-assignment)
(assert q)
(check-sat)
(get-model)
(get-assignment)
(exit)
