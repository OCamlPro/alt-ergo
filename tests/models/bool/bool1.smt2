(set-logic QF_UF)
; (set-option :produce-models true) ; enable model generation
(declare-const p Bool)
(declare-const q Bool)
; (declare-const t Int)
(define-fun nq () Bool q)
(assert (=> (not p) (not nq)))
(check-sat)
(get-model)
(get-assignment)
(assert q)
(check-sat)
(get-model)
(get-assignment)
(exit)
