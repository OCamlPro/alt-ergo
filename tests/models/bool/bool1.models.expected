
unknown
(
  (define-fun q () Bool false)
)
()


unknown
(
  (define-fun p () Bool true)
  (define-fun q () Bool true)
)
()

